src/lex.cpp

```diff
@@ -450,10 +450,34 @@ void lexer::parse_current_token() {
         }
 
       case '\n':
-      case '\r':
+      case '\r': {
+        char8* c_up_to_first_matching_quote = nullptr;
+        char8* c_end = c;
+        if (c_end[0] == '\r' && c_end[1] == '\n') {
+          c_end += 2;
+        } else {
+          c_end += 1;
+        }
+        for (;;) {
+          if (*c_end == opening_quote) {
+            if (c_up_to_first_matching_quote) {
+              break;
+            }
+            c_up_to_first_matching_quote = ++c_end;
+          } else if (*c_end == '\r' || *c_end == '\n' ||
+                     (*c_end == '\0' && this->is_eof(c_end))) {
+            if (c_up_to_first_matching_quote) {
+              c = c_up_to_first_matching_quote;
```

Comment:
(From comment on line 454)

This is what I mean by 'decrement':
```suggestion
            c_up_to_first_matching_quote = c_end;
            ++c_end;
          } else if (*c_end == '\r' || *c_end == '\n' ||
                     (*c_end == '\0' && this->is_eof(c_end))) {
            if (c_up_to_first_matching_quote) {
              c = c_up_to_first_matching_quote + 1;
```